core/chains/evm/client/errors.go

```diff
@@ -353,9 +361,15 @@ func (s *SendError) IsServiceUnavailable(configErrors *ClientErrors) bool {
 	return s.is(ServiceUnavailable, configErrors)
 }
 
-// IsOutOfCounters is a zk chain specific error returned if the transaction is too complex to prove on zk circuits
-func (s *SendError) IsOutOfCounters(configErrors *ClientErrors) bool {
-	return s.is(OutOfCounters, configErrors)
+// IsTerminallyStuck indicates if a transaction was stuck without any chance of inclusion
+func (s *SendError) IsTerminallyStuckConfigError(configErrors *ClientErrors) bool {
+	return s.is(TerminallyStuck, configErrors)
+}
+
+// Naming discrepancy is due to the generic transaction states introduced by ChainWriter
+// Errors classified as terminally stuck are considered fatal since the transaction payload should NOT be retried by external callers
+func (s *SendError) IsFatal() bool {
```

Comment:
I'm having trouble reading this doc. Let's rephrase this comment to be idiomatic by starting with the name of the method:
```suggestion
// IsFatal ...
func (s *SendError) IsFatal() bool {
```

Reply:
Updated in the latest commit. Hopefully it's easier to read